### DISCUSSION TITLE
Using got.get to make a REST API call in a cron triggered app

### DISCUSSION THREAD

**Discussion by pkassios** [deta/cloud-docs#584] [Help]
Hi everyone,

I am trying to deploy an app which I had working on Heroku before on deta.
It's a background process written in javascript which connects to a REST API and downloads some stuff.
For the HTTP Callout I am using the [got module](https://github.com/sindresorhus/got/blob/HEAD/documentation/quick-start.md) but it doesn't seem to work and I dont know why because there is nothing written in the logs..
My code looks like this

```javascript
app.lib.cron(event =>
{
  yesterday.setDate(yesterday.getDate() - 1)
  console.log('Yesterday: ' + yesterday);
  got.get(
    'https://someurl.com', {responseType: 'json'})
    .then(res => 
    {
      const headerDate = res.headers && res.headers.date ? res.headers.date : 'no response date';
      console.log('Status Code:', res.statusCode);
      console.log('Date in Response header:', headerDate);
      const orders = res.body.items;
  });
```
 
 The first log message is displayed but the second one not and I get nothing else in the logs..
 I am also catching any errors and logging them but no error is logged as well.

```javascript 
     .catch(err => {
        console.log('Error: ', err.message);
      })
 ```    

Any ideas? Is this something maybe not supported?

**Comment by lemonyte** [comment]
1. Please fix the formatting of your post so that we can read the code
2. Try using `console.error()` instead of `console.log()`, because `stdout` may not show up in logs, while `stderr` does.

**Reply by pkassios** [reply]
@LemonPi314 thanks a lot for the quick response. I fixed the format of the code in the post, sorry about this. I also tried console.error instead of console.log but still no error gets printed in the logs.

**Reply by lemonyte** [reply]
Are you checking the logs using Visor or the `deta logs` command?

**Reply by lemonyte** [reply]
Note that ES modules are not supported, and the runtime is node v14.

**Reply by pkassios** [reply]
@LemonPi314 I guess this is the reson, the Got Module is an ES module and requires min node v15 :-(

**Reply by lemonyte** [reply]
Perhaps `node-fetch` would be a suitable alternative.

**Reply by pkassios** [reply]
@LemonPi314 I switched to node-fetch and I am getting the same behavior. I am checking the logs both on deta site and using deta logs -f and I am only getting the log outpout before the fetch call..
It seems there is an issue making callouts? 
The code runs fine locally removing the app.lib.cron
The bad thing is that there is nothing in the logs to help out so unless you have any other ideas I will probably give up on deta..
Thanks again anyway!

```javascript
app.lib.cron(event =>
{
  yesterday.setDate(yesterday.getDate() - 1)
  console.error('Yesterday: ' + yesterday); --> This gets printed in the console
  fetch(
    '<URL goes here>')
    .then(res => res.json())
    .then(json =>
    {
      console.error("Inside fetch"); --> This doesn't get printed in the console...
...
```

**Reply by lemonyte** [reply]
You need to make your cron function asynchronous and use `await` with `fetch()`. Otherwise, a Promise is created, but the function returns and the Micro instance is killed before the Promise has a chance to run.

```js
const { app } = require('deta');
const fetch = require('node-fetch');

app.lib.cron(async (event) => {
  const response = await fetch("<URL goes here>")
  const data = await response.json();
  console.error(data)
  return data;
});

module.exports = app;
```
`node-fetch` version `2.6.7`
